python/sanlock.c

```diff
@@ -136,14 +190,16 @@ __parse_resource(PyObject *obj, struct sanlk_resource **res_ret)
 
         }
 
-        if (!PyArg_ParseTuple(disk, "sK", &path, &offset)) {
+        if (!PyArg_ParseTuple(disk, "O&K", pypath_converter, &path, &offset)) {
             /* Override the error since it confusing in this context. */
             set_error(PyExc_ValueError, "Invalid disk %s", disk);
+            Py_XDECREF(path);
             goto exit_fail;
         }
 
-        strncpy(res->disks[i].path, path, SANLK_PATH_LEN - 1);
+        strncpy(res->disks[i].path, PyBytes_AsString(path), SANLK_PATH_LEN - 1);
         res->disks[i].offset = offset;
+        Py_DECREF(path);
```

Comment:
And here?

And not add a finally label for cleanup?

Reply:
see prev comment.